src/vault.cairo

```diff
@@ -156,5 +178,46 @@ mod Vault {
     
             self.emit(LiquidityWithdrawn { user, token, amount });
         }
+
+        fn add_deposit_contract(ref self: ContractState, deposit_contract: ContractAddress){
+            let user = get_caller_address();
+            assert(deposit_contract.is_non_zero(), 'Deposit contract address is zero');
+            self.activeContracts.entry(user).write(deposit_contract);
+            self.emit(ContractAdded {user, deposit_contract});
+        }
+
+        fn protect_position(
+            self: @TContractState, 
+            deposit_contract: ContractAddress, 
+            user: ContractAddress, 
+            amount: TokenAmount
+        ) {
+            let token = self.token.read();
+            let caller = get_caller_address();
+            let vault_contract = get_contract_address();
+            let vault_owner = self.ownable.Ownable_owner.read();
```

Comment:
Please, use the [function](https://docs.openzeppelin.com/contracts-cairo/0.19.0/api/access#OwnableComponent-two-step-owner)